format/avro/decoders/fixed.go

```diff
@@ -0,0 +1,25 @@
+package decoders
+
+import (
+	"errors"
+	"github.com/wader/fq/pkg/scalar"
+
+	"github.com/wader/fq/format/avro/schema"
+	"github.com/wader/fq/pkg/decode"
+)
+
+func decodeFixedFn(schema schema.SimplifiedSchema, sms ...scalar.Mapper) (DecodeFn, error) {
+	if schema.Size < 0 {
+		return nil, errors.New("fixed size must be greater than or equal to zero")
+	}
+	size := int64(schema.Size)
+	// Fixed instances are encoded using the number of bytes declared in the schema.
+	return func(name string, d *decode.D) interface{} {
+		bb := d.FieldRawLen(name, size*8, sms...)
```

Comment:
Some reason this is raw and not `d.FieldU(nane, size*8, sms)` etc?

Reply:
Fixed is a byte array of a known size. Can be larger than 8 bytes so I don't think FieldU is the correct type here. Let me know if I'm misunderstanding something here.